crates/sui-protocol-config/src/lib.rs

```diff
@@ -1403,15 +1397,17 @@ impl ProtocolConfig {
     pub fn set_zklogin_auth(&mut self, val: bool) {
         self.feature_flags.zklogin_auth = val
     }
-    pub fn set_max_tx_gas_for_testing(&mut self, max_tx_gas: u64) {
+    pub fn set_upgraded_multisig_for_testing(&mut self, val: bool) {
+        self.feature_flags.upgraded_multisig_supported = val
+    }
+    #[allow(non_snake_case)]
+    pub fn set_max_tx_gas_for_testing_UNSAFE(&mut self, max_tx_gas: u64) {
```

Comment:
How come these ones are labelled `UNSAFE` (compared to the other test settings)? AFAICT it's generally inadvisable (i.e. unsafe) to override a protocol config outside of a test or debug scenario.

Also, from what I can tell, the accessor function is still going to introduce a `set_max_tx_gas_for_testing`, is that right? If so, does this function exist because the other one is only available if `debug_assertions` are enabled? That in itself struck me as strange -- it should be reasonable to run tests in in release mode, and this seems to break that.

Curious to get other people's thoughts on this (@mystenmark in particular), but to my taste, having different functions available in different build configs is more likely to cause confusion and therefore bugs than having `_for_testing` functions generally available, and relying on code reviews to ensure that we don't use these outside of test scenarios.

Reply:
> How come these ones are labelled UNSAFE (compared to the other test settings)? AFAICT it's generally inadvisable (i.e. unsafe) to override a protocol config outside of a test or debug scenario.

Agreed. The problem here is these functions were being called outside of debug/test context (particularly in adapter transactional test runner).
The proper thing to do is update the callers to only use then in debug/test mode but thats a bit more work I was punting.

> Also, from what I can tell, the accessor function is still going to introduce a set_max_tx_gas_for_testing, is that right? If so, does this function exist because the other one is only available if debug_assertions are enabled? 

Yes and yes.

> That in itself struck me as strange -- it should be reasonable to run tests in in release mode, and this seems to break that.

True, whats the right way of guarding these functions then?

> but to my taste, having different functions available in different build configs is more likely to cause confusion and therefore bugs than having _for_testing functions generally available, and relying on code reviews to ensure that we don't use these outside of test scenarios.

Yes for simplicity, we can make it generally available and not compiled out. Either way is fine by me.